src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1228,10 +1228,12 @@ void MacroAssembler::lookup_interface_method_stub(Register recv_klass,
   add(recv_klass, recv_klass, vtable_start_offset + ioffset);
   // itableOffsetEntry[] itable = recv_klass + Klass::vtable_start_offset() + sizeof(vtableEntry) * recv_klass->_vtable_len;
   // temp_itbl_klass = itable[0]._interface;
-  ldr(temp_itbl_klass, Address(recv_klass, scan_temp, Address::lsl(exact_log2(vtableEntry::size_in_bytes()))));
+  int vtblEntrySize = vtableEntry::size_in_bytes();
+  assert(vtblEntrySize == wordSize, "ldr shift amount must be lsl#3");
```

Comment:
This assert relies on that `vtblEntrySize` is the same in debug and release builds.
In theory, the assert can be true  in a debug build and false in a release build.
As the `guarantee` in `encode` will work for both debug and release builds, having the assert here looks redundant.
If we want to have an assert here, I'd use the `static_assert`:
```
static_assert(sizeof(vtableEntry) == wordSize, "ldr lsl shift amount must be 3");
```

BTW, "lsl#3" in the error message "ldr shift amount must be lsl#3" looks confusing to me.